crates/chunk/examples/chunk_add_nivc.rs

```diff
@@ -0,0 +1,304 @@
+use arecibo::supernova::snark::CompressedSNARK;
+use arecibo::supernova::{
+    NonUniformCircuit, PublicParams, RecursiveSNARK, StepCircuit, TrivialSecondaryCircuit,
+};
+use arecibo::traits::snark::default_ck_hint;
+use arecibo::traits::{CurveCycleEquipped, Dual, Engine};
+use bellpepper_chunk::traits::{ChunkCircuitInner, ChunkStepCircuit};
+use bellpepper_chunk::{FoldStep, InnerCircuit};
+use bellpepper_core::num::AllocatedNum;
+use bellpepper_core::{ConstraintSystem, SynthesisError};
+use ff::{Field, PrimeField};
+use flate2::write::ZlibEncoder;
+use flate2::Compression;
+use halo2curves::bn256::Bn256;
+use std::marker::PhantomData;
+use std::time::Instant;
+
+pub type E1 = arecibo::provider::Bn256EngineKZG;
+pub type E2 = arecibo::provider::GrumpkinEngine;
+pub type EE1 = arecibo::provider::hyperkzg::EvaluationEngine<Bn256, E1>;
+pub type EE2 = arecibo::provider::ipa_pc::EvaluationEngine<E2>;
+// SNARKs without computation commitments
+pub type S1 = arecibo::spartan::batched::BatchedRelaxedR1CSSNARK<E1, EE1>;
+pub type S2 = arecibo::spartan::snark::RelaxedR1CSSNARK<E2, EE2>;
+// SNARKs with computation commitments
+pub type SS1 = arecibo::spartan::batched_ppsnark::BatchedRelaxedR1CSSNARK<E1, EE1>;
+pub type SS2 = arecibo::spartan::ppsnark::RelaxedR1CSSNARK<E2, EE2>;
+
+/*******************************************************
+ * User side
+ *******************************************************/
+
+// Used to instantiate our utility
+#[derive(Clone, Eq, PartialEq, Debug)]
+struct ChunkStep<F: PrimeField> {
```

Comment:
The main issue here is this is a Nova integration, and we probably want a SuperNova integration instead. A good place to start from is our Supernova benches at https://github.com/lurk-lab/arecibo/blob/dev/benches/common/supernova/bench.rs#L26

We can of course look at that in a later PR.